From 0da3cd83c43f6dc06f58326410d90b66722459d8 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Tue, 30 Jan 2024 06:27:38 -0500
Subject: [PATCH] bananapi m2s support fixup

Allow meson-g12b-s922x-bananapi-m2s.dtb to be compiled.

HDMI doesn't come up until handing off to the kernel, at
which point it only displays in green and magenta. Removing
the default *-u-boot.dtsi and adding one specific to each variant
corrects the issue.

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 arch/arm/dts/Makefile                                      | 1 +
 ...boot.dtsi => meson-g12b-a311d-bananapi-m2s-u-boot.dtsi} | 0
 arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi     | 7 +++++++
 3 files changed, 8 insertions(+)
 rename arch/arm/dts/{meson-g12b-bananapi-u-boot.dtsi => meson-g12b-a311d-bananapi-m2s-u-boot.dtsi} (100%)
 create mode 100644 arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 9d28a485be..8559c70810 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -245,6 +245,7 @@ dtb-$(CONFIG_ARCH_MESON) += \
 	meson-g12b-odroid-n2l.dtb \
 	meson-g12b-odroid-n2-plus.dtb \
 	meson-g12b-radxa-zero2.dtb \
+	meson-g12b-s922x-bananapi-m2s.dtb \
 	meson-sm1-bananapi-m2-pro.dtb \
 	meson-sm1-bananapi-m5.dtb \
 	meson-sm1-khadas-vim3l.dtb \
diff --git a/arch/arm/dts/meson-g12b-bananapi-u-boot.dtsi b/arch/arm/dts/meson-g12b-a311d-bananapi-m2s-u-boot.dtsi
similarity index 100%
rename from arch/arm/dts/meson-g12b-bananapi-u-boot.dtsi
rename to arch/arm/dts/meson-g12b-a311d-bananapi-m2s-u-boot.dtsi
diff --git a/arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi b/arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi
new file mode 100644
index 0000000000..236f2468dc
--- /dev/null
+++ b/arch/arm/dts/meson-g12b-s922x-bananapi-m2s-u-boot.dtsi
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2019 BayLibre, SAS.
+ * Author: Neil Armstrong <narmstrong@baylibre.com>
+ */
+
+#include "meson-g12-common-u-boot.dtsi"
-- 
2.39.2

